### ISSUE TITLE
[Bug]: Narrator is not announcing the label name of the checkbox in “16 passes” pane. 

### ISSUE THREAD

**Issue by msftedad** [storybookjs/storybook#24125]
### Describe the bug

Narrator is not announcing the label name of the checkbox in “16 passes” pane. 

Actual Result:
Narrator is not announcing the label name of the checkbox in “16 passes” pane.

Expected Result:
Narrator should announce the label of the checkbox in “16 passes” pane.

### To Reproduce

1. Open URL: [Common Navigation - Scrolling Panel ⋅ Storybook (windows.net)](https://vivacommonstorybooktest.blob.core.windows.net/$web/index.html?path=/story/common-navigation--scrolling-panel&globals=backgrounds.grid:!false;backgrounds.value:transparent)
2.  Navigate to “App bar” button at right side of the pane invoke it and navigate to “Default” link invoke it. 
3. Now navigate to “Accessibility” tab item invoke it and navigate to “16 passes” button invoke it and navigate check boxes and observe the narrator announcement. 

Attachments:

https://github.com/storybookjs/storybook/assets/93735775/bf9cc9d8-f243-43f4-acd5-0d9089f830a6

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          

### System

```bash
OS: Windows 11
OS version: 22H2 (OS Build 22621.2134)
Edge Version: 116.0.1938.69 (Official build) (64-bit)
```


### Additional context

_No response_

**Comment by Nevnet99** [comment]
I can take a look at this also :)

**Comment by Nevnet99** [comment]
It seems like with the current functionality this is not an issue, 

no matter which checkbox you click on it enables all the other checkboxes anyway so the current label "Highlight label" is correct. 

However this does pose the question, should these checkboxes be individually clickable to only highlight specific errors 🤔
or should the checkboxes highlight everything no matter what level of nesting they are clicked on? 

I've attached a video of the current functionality I assume this is wrong and a larger bug ticket is blocking this: 
https://github.com/storybookjs/storybook/assets/28801236/6862124f-eede-4cfc-93ed-7dcf60b348d6

**Comment by Nevnet99** [comment]
Comment on #24126 this could be solved with this PR

**Comment by yannbf** [comment]
Hey @msftedad thank you so much for reporting this!

> No matter which checkbox you click on it enables all the other checkboxes anyway so the current label "Highlight label" is correct.
@Nevnet99 this is actually not correct. The behavior of the addon panel is to highlight the offending element. Because, in your example, both rules apply to the same element, then both are toggled, given that the element is highlighted. 
![2023-10-12 17 47 38](https://github.com/storybookjs/storybook/assets/1671563/d632a2db-60a3-4959-9c03-c8c493b08426)

I think it's up for discussion wether the checkboxes should all check, as long as they're targeting the same elements, but I believe it was a design decision from the past.

**Comment by msftedad** [comment]
Hi team, Any update on this issue?

**Comment by yannbf** [comment]
Hey @msftedad I don't think so. We are quite busy preparing Storybook 8 beta, so if you'd like to take a stab at contributing, we'd be thankful for that!

**Comment by Sidnioulz** [comment]
Hi folks,

This part of the UI has been removed. The highlight button now works globally instead. Issues related to the highlight button's own ARIA markup should be handled through https://github.com/storybookjs/storybook/pull/31702 and there no longer is any checkbox. The button will be a switch with an aria-checked status going forward.
